File: .tmp_py4013/batch_11_dump/3270_find_the_key_of_the_numbers.js
// LeetCode 3270 - Find the Key of the Numbers
// https://leetcode.com/problems/find-the-key-of-the-numbers/

var generateKey = function(num1, num2, num3) {
    let ans = 0, mul = 1;
    for (let t = 0; t < 4; t++) {
        const d = Math.min(num1 % 10, num2 % 10, num3 % 10);
        ans += d * mul;
        mul *= 10;
        num1 = Math.floor(num1 / 10); num2 = Math.floor(num2 / 10); num3 = Math.floor(num3 / 10);
    }
    return ans;
};

File: .tmp_rs4013/rust_check15/3270_find_the_key_of_the_numbers.rs
struct Solution;
// LeetCode 3270 - Find the Key of the Numbers
// https://leetcode.com/problems/find-the-key-of-the-numbers/

impl Solution {
    pub fn generate_key(mut num1: i32, mut num2: i32, mut num3: i32) -> i32 {
        let mut ans = 0;
        let mut mul = 1;
        for _ in 0..4 {
            let d = num1 % 10;
            let d = d.min(num2 % 10).min(num3 % 10);
            ans += d * mul;
            mul *= 10;
            num1 /= 10;
            num2 /= 10;
            num3 /= 10;
        }
        ans
    }
}

fn main() {}
